MekHQ/src/mekhq/campaign/ResolveScenarioTracker.java

```diff
@@ -183,6 +183,10 @@ public void processGame() {
             if (!e.getSubEntities().isEmpty()) {
                 // Sub-entities have their own entry in the VictoryEvent data
                 continue;
+            } else if ("-1".equals(e.getExternalIdAsString())) {
```

Comment:
I'd almost be tempted to say that if it doesn't parse to a UUID, then we reject it.

Reply:
Not sure if -1 parses to a UUID, though.

Reply:
It doesn't, hence the check if it is the external id and writing it as an error before continuing. The exception was trying to turn -1 into a UUID.

Reply:
Right, in that case I'd suggest that we be a little more proactive about it and reject the incoming entity if the external ID doesn't parse to a UUID.

Alternately, maybe we want to generate a new UUID for it in that case. Otherwise, what'll happen is that entities added mid-game won't show up in the scenario resolution dialog at all, and I don't know whether that's worse or better than a "new" entity showing up.

Reply:
Currently, I'm thinking it's some kind of deserialization scenario, because deserialization sometimes doesn't seem to call a constructor.

Reply:
MegaMek should be generating a UUID for all external ids... I have no clue how it can possibly be -1 given that is the case, but it is.

That's my best guess too, but I can only take it based on the logs.

Reply:
This just prevents it from being processed (hence the continue). I thought it better to just skip processing given our internal system has no idea what the unit is instead of processing a ghost unit